system-test/stake-operations-testcases/offline_stake_operations.sh

```diff
@@ -0,0 +1,263 @@
+online_system_account_keypair=online_system_account_keypair.json
+
+solana-keygen new -o "$online_system_account_keypair" --no-passphrase --force --silent
+solana-keygen new -o "$online_nonce_account_keypair" --no-passphrase --force --silent
+
+online_system_account_pubkey="$(solana-keygen pubkey $online_system_account_keypair)"
+nonce_account_pubkey="$(solana-keygen pubkey $online_nonce_account_keypair)"
+
+# System account funding the stake account is offline, and the auth staker and withdrawer keypairs are offline
+offline_system_account_keypair=offline_system_account_keypair.json
+offline_staker_keypair=offline_staker_keypair.json
+offline_withdrawer_keypair=offline_withdrawer_keypair.json
+offline_custodian_keypair=offline_custodian_keypair.json
+
+solana-keygen new -o "$offline_system_account_keypair" --no-passphrase --force --silent
+solana-keygen new -o "$offline_staker_keypair" --no-passphrase --force --silent
+solana-keygen new -o "$offline_withdrawer_keypair" --no-passphrase --force --silent
+solana-keygen new -o "$offline_custodian_keypair" --no-passphrase --force --silent
+
+offline_system_account_pubkey="$(solana-keygen pubkey $offline_system_account_keypair)"
+offline_withdrawer_pubkey="$(solana-keygen pubkey $offline_withdrawer_keypair)"
+offline_staker_pubkey="$(solana-keygen pubkey $offline_staker_keypair)"
+offline_custodian_pubkey="$(solana-keygen pubkey $offline_custodian_keypair)"
+
+# Airdrop some funds to the offline account.
+solana airdrop 100 $offline_system_account_pubkey
+solana airdrop 2 $online_system_account_pubkey
+
+# Create a nonce account funded by the online account, with the authority given to the offline account
+solana create-nonce-account $online_nonce_account_keypair 1 --nonce-authority $offline_system_account_pubkey --keypair $online_system_account_keypair
+nonce="$(solana nonce $nonce_account_pubkey)"
+
+execution_step OFFLINE SYSTEM ACCOUNT BALANCE BEFORE CREATING STAKE ACCOUNTS
+(
+  set -x
+  solana balance $offline_system_account_pubkey
+)
+
+################################
+execution_step CREATE OFFLINE STAKE ACCOUNT
+################################
+
+# Create a stake account funded by the offline system account
+
+stake_account_keypair=stake_account_keypair.json
+solana-keygen new -o "$stake_account_keypair" --no-passphrase --force --silent
+stake_account_address="$(solana-keygen pubkey $stake_account_keypair)"
+
+sign_only="$(solana create-stake-account $stake_account_keypair 50 \
+  --sign-only --blockhash $nonce --nonce $nonce_account_pubkey --nonce-authority $offline_system_account_keypair \
+  --stake-authority $offline_staker_pubkey --withdraw-authority $offline_withdrawer_pubkey \
+  --custodian $offline_custodian_pubkey \
+  --lockup-epoch 999 \
+  --keypair $offline_system_account_keypair --url http://0.0.0.0)"
+
+signers=()
+while read LINE; do
+  signers+=( --signer $LINE )
+done <<<"$(sed -Ee $'s/^  ([a-zA-Z0-9]+=[a-zA-Z0-9]+)/\\1/\nt\nd' <<<"$sign_only")"
```

Comment:
Might factor this out to a function and tighten the regex to the base58 alphabet

To get around the "bash can't return an array" thing, you can build up a string instead

Reply:
sg.  Will follow up with this change.